building_dialouge_webapp/heat/tables.py

```diff
@@ -1,17 +1,57 @@
+from math import isclose
+
 import pandas as pd
 
 
+def format_currency(value):
+    """Format a number as currency (Euro)."""
+    if value is None:
+        return ""
+    if isinstance(value, (int, float)):
+        if isclose(value, round(value)):
+            return f"{int(value):,}".replace(",", ".") + " €"
+        return f"{value:,.2f}".replace(",", "X").replace(".", ",").replace("X", ".") + " €"
+    return str(value)
+
+
 class Table:
+    """Base class for generating and formatting tables."""
+
     translations = {
-        "procedure": "Maßnahme",
         "scenario1": "Szenario 1",
         "scenario2": "Szenario 2",
         "scenario3": "Szenario 3",
+    }
+
+    def __init__(self, data):
+        self.data = data
+
+    def translate_column(self, column_name):
+        return self.translations.get(column_name, column_name)
+
+    def to_html(self, dataframe, title):
+        html_table = dataframe.to_html(classes="table", escape=False, index=False)
+        html_table = html_table.replace('<table border="1" class="dataframe table">', f'<table class="{title}">')
+
+        for header, translated in self.translations.items():
+            html_table = html_table.replace(
+                f"<th>{translated}</th>",
+                f'<th class="{header}"><span>{translated}</span></th>',
+            )
```

Comment:
Using `Styler` (see example `a.style`), we can change styling of table using pandas operators instead of using str-functions.
See https://pandas.pydata.org/docs/user_guide/style.html#Table-Styles for examples.
Maybe https://pandas.pydata.org/docs/user_guide/style.html#Data-Cell-CSS-Classes could solve this.
But this has no prio! String manipulation works, so we could go on with it as well. Just a hint.